bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorImpl.java

```diff
@@ -49,29 +46,29 @@
  * @author Chris Jackson - Handle checks on multiple selection parameters
  */
 @Component
+@NonNullByDefault
 public final class ConfigDescriptionValidatorImpl implements ConfigDescriptionValidator {
 
-    private final Logger logger = LoggerFactory.getLogger(ConfigDescriptionValidatorImpl.class);
-
-    private ConfigDescriptionRegistry configDescriptionRegistry;
-    private TranslationProvider translationProvider;
+    private static final List<ConfigDescriptionParameterValidator> VALIDATORS = List.of( //
+            ConfigDescriptionParameterValidatorFactory.createRequiredValidator(), //
+            ConfigDescriptionParameterValidatorFactory.createTypeValidator(), //
+            ConfigDescriptionParameterValidatorFactory.createMinMaxValidator(), //
+            ConfigDescriptionParameterValidatorFactory.createPatternValidator() //
+    );
 
-    private static final List<ConfigDescriptionParameterValidator> VALIDATORS = Collections
-            .unmodifiableList(Arrays.asList(ConfigDescriptionParameterValidatorFactory.createRequiredValidator(),
-                    ConfigDescriptionParameterValidatorFactory.createTypeValidator(),
-                    ConfigDescriptionParameterValidatorFactory.createMinMaxValidator(),
-                    ConfigDescriptionParameterValidatorFactory.createPatternValidator()));
+    private final Logger logger = LoggerFactory.getLogger(ConfigDescriptionValidatorImpl.class);
 
-    private BundleContext bundleContext;
+    private final ConfigDescriptionRegistry configDescriptionRegistry;
+    private final TranslationProvider translationProvider;
+    private final BundleContext bundleContext;
 
     @Activate
-    protected void activate(BundleContext bundleContext) {
+    public ConfigDescriptionValidatorImpl(final @Reference BundleContext bundleContext,
```

Comment:
You can probably fix the build by removing `@Reference` here:

```suggestion
    public ConfigDescriptionValidatorImpl(final BundleContext bundleContext,
```

The `BundleContext` is not something you can get a reference on like another component. But I think the annotations processor is smart enough to inject the `BundleContext` when it sees such a parameter. We also use it like this with the `StartLevelService`.

Reply:
Thanks for the hint. I applied your suggestion.